tests/testthat/test-submodule.R

```diff
@@ -93,43 +93,47 @@ test_that("Can install a repo with a submodule", {
       "-c", "user.name=foobar", "-c", paste0("user.email=", shQuote("<>")),
       "commit", "-m", shQuote("Initial commit")))
   })
-  module <- file.path("submodule", ".gitmodules")
-  on.exit(unlink(module), add = TRUE)
+
+  sub <- tempfile()
+  dir.create(sub)
+  on.exit(unlink(sub,recursive=TRUE,force=TRUE),add=TRUE)
+
+  module <- file.path(sub, ".gitmodules")
 
   writeLines(con = module,
     sprintf(
 '[submodule "foo"]
 	path = R
 	url = file://%s
-	branch = master
 [submodule "bar"]
 	path = bar
-	url = file://%s
-	branch = master',
+	url = file://%s',
       URLencode(dir),
       URLencode(dir)
     )
   )
 
   # The bar submodule is in .Rbuildignore, so we will not fetch it
-  build_ignore <- file.path("submodule", ".Rbuildignore")
-  on.exit(unlink(build_ignore), add = TRUE)
-
+  build_ignore <- file.path(sub, ".Rbuildignore")
```

Comment:
Is this needed? You at least should keep the `on.exit()`, otherwise temporary files are not cleaned up.

Reply:
The temporary files contained in `sub` would also be deleted when `sub` is deleted in the previous `on.exit()`